src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogTests.cs

```diff
@@ -15,62 +16,314 @@ public void EventLogReinitializationException()
+        {
+            string source = "Source_" + nameof(OverflowAndRetention_Set);
+            string log = "Overflow_Set";
+            int retentionDays = 30; //A number between 0 and 365 should work
+
+            try
+            {
+                EventLog.CreateEventSource(source, log);
+                using (EventLog eventLog = new EventLog())
+                {
+                    eventLog.Source = source;
+
+                    //The second argument is only used when the overflow policy is set to OverWrite Older
+                    eventLog.ModifyOverflowPolicy(OverflowAction.OverwriteOlder, retentionDays);
+                    Assert.Equal(OverflowAction.OverwriteOlder, eventLog.OverflowAction);
+                    Assert.Equal(retentionDays, eventLog.MinimumRetentionDays);
+                }
+            }
+            finally
+            {
+                EventLog.DeleteEventSource(source);
+                EventLog.Delete(log);
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void OverflowAndRetentionDaysOutOfRange()
+        {
+            using (EventLog eventLog = new EventLog())
+            {
+                eventLog.Log = "Application";
+                Assert.Throws<ArgumentOutOfRangeException>(() => eventLog.ModifyOverflowPolicy(OverflowAction.OverwriteOlder, 400));
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void MachineName_Set()
+        {
+            string source = "Source_" + nameof(MachineName_Set);
+            using (EventLog eventLog = new EventLog())
+            {
+                eventLog.Log = "Application";
+                eventLog.MachineName = Environment.MachineName;
+                try
+                {
+                    EventLog.CreateEventSource(source, eventLog.LogDisplayName);
+                    Assert.True(EventLog.SourceExists(source, Environment.MachineName));
+                }
+                finally
+                {
+                    EventLog.DeleteEventSource(source);
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void RegisterDisplayLogName()
+        {
+            string log = "DisplayName";
+            string source = "Source_" + nameof(RegisterDisplayLogName);
+            string messageFile = GetTestFilePath();
+            long DisplayNameMsgId = 45;
+            EventSourceCreationData sourceData = new EventSourceCreationData(source, log);
+
+            try
+            {
+                EventLog.CreateEventSource(sourceData);
+                log = EventLog.LogNameFromSourceName(source, ".");
+                using (EventLog eventLog = new EventLog(log, ".", source))
+                {
+                    if (messageFile.Length > 0)
```

Comment:
I don't understand this check -- you haven't written to the file yet, this will always be false

Reply:
MessageFile is a string containing the name of the file